python/ray/util/client/__init__.py

```diff
@@ -69,7 +70,7 @@ def connect(self,
                 metadata=metadata,
                 connection_retries=connection_retries)
             self.api.worker = self.client_worker
-            self.client_worker._server_init(job_config)
+            self.client_worker._server_init(job_config, _skip_uploading)
```

Comment:
Instead of adding a separate flag, could we always skip "upload" if job_config is None?